sql/core/src/test/scala/org/apache/spark/sql/GeneratorFunctionSuite.scala

```diff
@@ -91,7 +91,7 @@ class GeneratorFunctionSuite extends QueryTest with SharedSQLContext {
     val df = Seq((1, Seq(1, 2, 3)), (2, Seq())).toDF("a", "intList")
     checkAnswer(
       df.select(explode_outer('intList)),
-      Row(1) :: Row(2) :: Row(3) :: Nil)
+      Row(1) :: Row(2) :: Row(3) :: Row(null) :: Nil)
```

Comment:
So I think we should move away from that definition, and make `outer` independent from `join`. The reason for this is that I think it is super confusing that the following queries produce different results:
```
df.select($"k", outer_explode($"v")) 
df.select(outer_explode($"v")) 
```
If you do not want the outer semantic, then just use a regular generator.